### PR TITLE
boot: some improvements to CatInitrds

### PR DESCRIPTION
**Pull request by hugelgupf**
I wanted to make sure that when you embed a concatenated Initrd into a LinuxImage, it'd show the right name if you call the Stringer or the Label() function.

### CODE REVIEW

File: pkg/boot/initrd.go [hunk 1]
```diff
@@ -7,22 +7,31 @@ package boot
...
-	return uio.NewLazyOpenerAt("", func() (io.ReaderAt, error) {
+// CatInitrds concatenates initrds on first ReadAt call from a list of
+// io.ReaderAts, pads them to a 512 byte boundary.
+func CatInitrds(initrds ...io.ReaderAt) io.ReaderAt {
+	var names []string
+	for _, initrd := range initrds {
+		names = append(names, stringer(initrd))
+	}
+
+	return uio.NewLazyOpenerAt(strings.Join(names, ", "), func() (io.ReaderAt, error) {
```
Comment (orangecms) [inline pkg/boot/initrd.go:23]: Now this makes sense, I was a bit puzzled about the string there and left it for later. :)

File: pkg/boot/linux_test.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+		{
+			desc: "curl lazy file",
+			img: &LinuxImage{
+				Kernel: httpKernel,
+				Initrd: CatInitrds(
+					httpInitrd1,
+					httpInitrd2,
+				),
+			},
+			want: "Linux(kernel=http://127.0.0.1/kernel, initrd=http://127.0.0.1/initrd1, http://127.0.0.1/initrd2)",
```
Comment (orangecms) [inline pkg/boot/linux_test.go:84]: Looking at this specific example, I would suggest to either prefix every initrd with `initrd=`, use a different separator than `, ` for the initrds, possibly just removing the space so it's clearer that they belong together, or a different separator between kernel and initrds, e.g., a semicolon. The first and second option would be more consistent with other notations like the kernel command-line and syslinux' (dropping the space, like in https://wiki.syslinux.org/wiki/index.php?title=Config#INITRD), respectively. Maybe I am overthinking it though.

Comment (hugelgupf) [reply pkg/boot/linux_test.go:84]: good point, i did both

**Comment by hugelgupf** [comment]
ptal

**Comment by hugelgupf** [comment]
Thanks, @orangecms!

**Pull request merged**